tasks/internal/update-fbc-catalog-task/update-fbc-catalog-task.yaml

```
@@ -126,6 +126,14 @@ spec:
             jq --arg fbc_fragment "${fbc_fragment}" \
               '[.items[] |select(.fbc_fragment==$fbc_fragment and .state!="failed")][0] // empty')
 
+          # if a previous build is found, but from_index and index_image are different, the result of the next
+          # expression should be different than 1, meaning that the previous build was a staging index build,
+          # then we should return here, causing the task to call IIB again, but with overwrite_from_index set
+          # to true.
+          if [ "$(jq '[ .from_index, .index_image ] | unique | length' <<< "${build}")" != 1 ]; then
```

Why not simply check if they are equal?
```suggestion
          if [ "$(jq '.from_index == .index_image' <<< "${build}")" != true ]; then
```

---

Yes. That makes more sense. I just adjusted as well the comparison to check if `.from_index != .index_image`  equals `true`  facilitate the understand following the logic flow (at least for me :D).

---

My thought was that if there's any uncertainty, we should err on the side of starting a new IIB task. So for any other result than specifically that those two fields are equal, do the exit/return.

But it seems this jq query will really never return anything other than "true" or "false", so what you used should be equivalent of my suggestion. For example, if neither is even set, it will return "true". And if one is set and the other not, it will return "false". So probably ok.